src/pages/Projects/components/RoleList/RoleList.tsx

```diff
@@ -1,85 +1,53 @@
+import Button from "../../../../components/Button";
+import RoleModal from "../RoleModal";
+import { Project, Role, useSkills, useRoles } from "../../../../services/api";
+
+type NewRole = Partial<Omit<Role, "id">>;
+
+interface RoleListProps {
+  project: Project;
+  createRole: (data: NewRole) => Promise<string | undefined>;
+  destroyRole: (roleId: string) => Promise<void>;
+}
 
 export default function RoleList({
   project,
   createRole,
   destroyRole,
-}: {
-  project: Project;
-  createRole: (data: Partial<Omit<Role, "id">>) => Promise<string | undefined>;
-  destroyRole: (roleId: string) => Promise<void>;
-}): JSX.Element {
+}: RoleListProps): JSX.Element {
+  const skills = useSkills();
+  const roles = useRoles({
+    filter: { project_id: { $eq: project.id } },
+    include: ["skills"],
+  });
   const [projectToEdit, setProjectToEdit] = useState<Project | null>(null);
   const [roleToDelete, setRoleToDelete] = useState<Role | null>(null);
-  const skills = useSkillsDefault();
-
-  const lastRoleIndex = Array.isArray(project?.roles)
-    ? project?.roles.length - 1
-    : -1;
+  const lastRoleIndex = Array.isArray(roles) ? roles.length - 1 : -1;
 
   return (
     <>
       <Button mb={4} onClick={() => setProjectToEdit(project)}>
         Add Role
       </Button>
 
-      {!isEmpty(project?.roles) && (
-        <>
-          <Box maxHeight="80vh" overflowY="auto">
-            <Table>
-              <Thead py={4}>
-                <Tr>
-                  <Th color="gray.800" textStyle="table.title">
-                    Roles
-                  </Th>
-                  <Th color="gray.800" textStyle="table.title">
-                    Start Date
-                  </Th>
-                  <Th color="gray.800" textStyle="table.title">
-                    Confidence
-                  </Th>
-                  <Th color="gray.800" textStyle="table.title">
-                    End Date
-                  </Th>
-                  <Th color="gray.800" textStyle="table.title">
-                    Confidence
-                  </Th>
-                  <Th color="gray.800" textStyle="table.title">
-                    Current Employees
-                  </Th>
-                  <Th
-                    py={4}
-                    pr={12}
-                    color="gray.800"
-                    textStyle="table.title"
-                    isNumeric
-                  >
-                    Actions
-                  </Th>
-                </Tr>
-              </Thead>
-              <Tbody>
-                {project?.roles?.map((role, index) => (
-                  <RoleListRow
-                    key={role.id}
-                    role={role}
-                    lastChild={lastRoleIndex === index}
-                    handleDeleteRole={setRoleToDelete}
-                  />
-                ))}
-              </Tbody>
-            </Table>
-          </Box>
-        </>
+      {isEmpty(roles) ? null : (
```

Comment:
I wonder if its not faster to do `!isEmpty(roles) && ...`

Reply:
I avoid using `&&` because I don't want falsy values to be rendered on the screen. Say `0`.

and do you mean faster as in execution time? if it is, it might be negligible. I'd rather go with the ternary and only refactor when I am able to measure how much of a difference it makes.

Reply:
no faster I just mean less code

Reply:
oh! got it. Yeah, I'd rather use a ternary anyway.